Backend/users/usersControllers.js

```diff
@@ -0,0 +1,88 @@
+const bcrypt = require("bcryptjs");
+const User = require("./usersModels");
+const { generateToken } = require("./usersHelper");
+const { ResponseErrorHandler } = require("../ErrorsHandler/ResponseErrorHandler");
+const { tryCatchHandler } = require("../ErrorsHandler/TryCatchHandler");
+
+module.exports.signup = async (req, res) => {
+  const tryFn = async () => {
+
+    const { email, username, password, isBlock } = req.body;
+    const hashedPassword = await bcrypt.hash(password, 12);
+
+    const credentials = {
+      username,
+      email,
+      password: hashedPassword
+    };
+
+    const emailExist = await User.findBy({
+      email: credentials.email,
+    });
+
+    const usernameExist = await User.findBy({
+      username: credentials.username,
+    });
+
+    if (emailExist || usernameExist) {
+
+      if (emailExist)
+        ResponseErrorHandler(res, 409, "invalid credentials")
+
+      else
```

Comment:
What is it?

Reply:
I will fix it 
if (existingUser.length !== 0) {
    return responseErrorHandler(res, 409, "User with provided email or username already exists");
  }